### PR TITLE
test(admin_audit): add test for user management listener 

### PR DESCRIPTION
**Pull request by kesselb**
* Resolves: technical debt

## Summary

The migration to event listener was done by https://github.com/nextcloud/server/pull/47865.

~Good bye hooks, hello event listener (at least for one case)~

~Test cases:~

~- Enable admin_audit~
~- Enable user => see log line in audit log~
~- Change email for user => see log line in audit log~

## TODO

- [x] CI
- [ ] Review
- [ ] Merge

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Comment by come-nc** [comment]
@kesselb conflicts :wink:

**Comment by kesselb** [comment]
@come-nc resolved :tada:

**Review by come-nc — APPROVED** [review]

File: apps/admin_audit/tests/Listener/UserManagementEventListenerTest.php [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+use OCA\AdminAudit\IAuditLogger;
+use OCA\AdminAudit\Listener\UserManagementEventListener;
+use OCP\IUser;
+use OCP\User\Events\UserChangedEvent;
+use PHPUnit\Framework\MockObject\MockObject;
+use Test\TestCase;
+
+class UserManagementEventListenerTest extends TestCase {
+	private IAuditLogger|MockObject $logger;
```
Comment (provokateurin) [inline apps/admin_audit/tests/Listener/UserManagementEventListenerTest.php:18]:
```suggestion
	private IAuditLogger&MockObject $logger;
```

File: apps/admin_audit/tests/Listener/UserManagementEventListenerTest.php [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+	private UserManagementEventListener $listener;
+
+	private MockObject|IUser $user;
```
Comment (provokateurin) [inline apps/admin_audit/tests/Listener/UserManagementEventListenerTest.php:22]:
```suggestion
	private IUser&MockObject $user;
```

**Review by provokateurin — APPROVED** [review]

**Pull request merged**